prov/efa/src/efa_shm.c

```diff
@@ -98,6 +98,13 @@ void efa_shm_info_create(const struct fi_info *app_info, struct fi_info **shm_in
 	int ret;
 	struct fi_info *shm_hints;
 
+	char *shm_provider;
+	if (rxr_env.use_sm2) {
+		shm_provider = "sm2";
+	} else {
+		shm_provider = "shm";
+	}
+
 	shm_hints = fi_allocinfo();
 	shm_hints->caps = FI_MSG | FI_TAGGED | FI_RECV | FI_SEND | FI_READ
```

Comment:
If I run this patch with current SM2 (in https://github.com/ofiwg/libfabric/pull/8781, fi_getinfo will return -61, because SM2 does not support some of the caps yet.